src/csharp/Microsoft.Spark/Sql/Types/Timestamp.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.Spark.Sql.Types
+{
+    /// <summary>
+    /// Represents Timestamp containing year, month, day, hour, minute, second, microsecond in
+    /// Coordinated Universal Time (UTC).
+    /// </summary>
+    public class Timestamp
+    {
+        private readonly DateTime _dateTime;
+        private static readonly DateTime s_unixTimeEpoch =
+            new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+
+        /// <summary>
+        /// Constructor for Timestamp class.
+        /// </summary>
+        /// <param name="dateTime">DateTime object</param>
+        public Timestamp(DateTime dateTime)
+        {
+            _dateTime = dateTime.ToUniversalTime();
+
+            if (dateTime.Kind != DateTimeKind.Utc)
+            {
+                throw new InvalidTimeZoneException(
+                    $"Invalid TimeZone for Timestamp, please use " +
+                    $"Coordinated Universal Time (UTC).");
+            }
+        }
+
+        /// <summary>
+        /// Constructor for Timestamp class, the defaut timezone is
+        /// Coordinated Universal Time (UTC).
+        /// </summary>
+        /// <param name="year">The year (1 through 9999)</param>
+        /// <param name="month">The month (1 through 12)</param>
+        /// <param name="day">The day (1 through the number of days in month)</param>
+        /// <param name="hour">The hour (0 through 23)</param>
+        /// <param name="minute">The minute (0 through 59)</param>
+        /// <param name="second">The second (0 through 59)</param>
+        /// <param name="microsecond">The microsecond (0 through 999999)</param>
+        public Timestamp(int year, int month, int day, int hour, int minute, int second, int microsecond)
+        {
+            if (microsecond <= 999999 && microsecond >= 0)
+            {
+                _dateTime = new DateTime(year, month, day, hour, minute, second)
+                    .AddTicks(microsecond * 10).ToUniversalTime();
```

Comment:
Shouldn't the given year, month, etc. already be in UTC? Why do we need to call `ToUniversalTime`? I think you need to pass `DateTimeKind` when you constructor the `DateTime` object.

Reply:
So pass `DateTimeKind` does not give the right time, still have 8 hours lack in this case so I have to convert it using `ToUniversalTime`.